src/main/java/io/stargate/sgv2/jsonapi/service/processor/HybridFieldExpander.java

```diff
@@ -0,0 +1,103 @@
+package io.stargate.sgv2.jsonapi.service.processor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import io.stargate.sgv2.jsonapi.api.model.command.Command;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandContext;
+import io.stargate.sgv2.jsonapi.api.model.command.impl.InsertManyCommand;
+import io.stargate.sgv2.jsonapi.api.model.command.impl.InsertOneCommand;
+import io.stargate.sgv2.jsonapi.config.constants.DocumentConstants;
+import io.stargate.sgv2.jsonapi.exception.ErrorCodeV1;
+import io.stargate.sgv2.jsonapi.util.JsonUtil;
+import java.util.Iterator;
+
+/**
+ * Helper class that encapsulates the logic for expanding "$hybrid" fields in Collection Documents
```

Comment:
nit: make it clear that this mutates the document in place.

Reply:
Will add if that is missing.

Reply:
Yes, line 21 has:

```Note, too, that input document(s) are modified in place ...
```